packages/legend-application/src/stores/AssistantService.ts

```diff
@@ -77,6 +79,12 @@ export class VirtualAssistantContextualDocumentationEntry {
   }
 }
 
+export enum SEARCH_MODE {
```

Comment:
Let's componentize this. This will be a component with:
1. Component: take a button trigger, produce a dropdown with the options
2. Store: `TextSearchAdvancedConfigState`: which stores all the needed states including the state to control if the dropdown should be hidden/shown and a method to modify search text

Reply:
We then can put this state as part of `AssistantService`